### ISSUE TITLE
Lint: Files missing copyright headers

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#8659]
<details><summary>Done</summary>

## Me
- [x] ActionBarOverflow.java
- [x] ActionButtons.java
- [x] ActionButtonStatus.java
- [x] CardAppearance.java
- [x] CardMarker.java
- [x] FileUtil.java
- [x] FunctionalInterfaces.java
- [x] HtmlUtils.java
- [x] ManuallyReportedException.java
- [x] NoSuchDeckException.java
- [x] Permissions.java
- [x] PreferenceExtensions.java
- [x] WebViewDebugging.java

## timrae
- [x] AsyncDialogFragment.java
- [x] CollectionLoader.java
- [x] ConfirmationDialog.java
- [x] ConfirmModSchemaException.java
- [x] DatabaseErrorDialog.java
- [x] DeckDropDownAdapter.java
- [x] DeckPickerBackupNoSpaceLeftDialog.java
- [x] DeckPickerConfirmDeleteDeckDialog.java
- [x] DeckPickerExportCompleteDialog.java
- [x] DeckPickerNoSpaceLeftDialog.java
- [x] DialogHandler.java
- [x] ExportDialog.java
- [x] FieldEditText.java
- [x] ImportDialog.java
- [x] IntentHandler.java
- [x] PopupMenuWithIcons.java
- [x] SimpleMessageDialog.java
- [x] StorageAccessException.java
- [x] SyncErrorDialog.java
- [x] UnknownHttpResponseException.java
- [x] VersionUtils.java

## Mike
- [x] AcraAnalyticsInteraction.java
- [x] AnalyticsDialogFragment.java
- [x] DeckPickerAnalyticsOptInDialog.java
- [x] ImportUtils.java
- [x] IntegerDialog.java

## @TarekkMA 
- [x] BundleUtils.java
- [x] CreateCustomStudySessionListener.java
- [x] CustomStudyDialogFactory.java

## hermanne
- [x] CardBrowserOrderDialog.java (#8698)

## Arthur
- [x] AbstractDeckTreeNode.java
- [x] AbstractSched.java
- [x] BooleanGetter.java
- [x] CardGetter.java
- [x] CardQueue.java
- [x] CollectionGetter.java
- [x] CollectionUtils.java
- [x] Conditional.java
- [x] Counts.java
- [x] DeckComparator.java
- [x] DeckDueTreeNode.java
- [x] DeckNameComparator.java
- [x] DeckTreeNode.java
- [x] EmptyNode.java
- [x] FilteredAncestor.java
- [x] LrnCard.java
- [x] LrnCardQueue.java
- [x] MathJax.java
- [x] NamedJSONComparator.java
- [x] NegatedConditional.java
- [x] PairWithBoolean.java
- [x] PairWithCard.java
- [x] ParsedNode.java
- [x] ParsedNodes.java
- [x] ProgressSenderAndCancelListener.java
- [x] Replacement.java
- [x] SimpleCardQueue.java
- [x] SingleTaskManager.java
- [x] StdModels.java
- [x] TaskListener.java
- [x] TaskListenerWithContext.java
- [x] TaskManager.java
- [x] TemplateError.java
- [x] TemplateFilters.java
- [x] Text.java
- [x] Tokenizer.java
- [x] Triple.java
- [x] UndoAction.java
- [x] WrongId.java

## iniju
- [x] AnkiFont.java
- [x] HtmlColors.java

## lukstbit
- [x] Constants.java
- [x] DirectCalendarInstanceUsage.java
- [x] DirectDateInstantiation.java
- [x] DirectGregorianInstantiation.java
- [x] DirectSystemCurrentTimeMillisUsage.java
- [x] DirectSystemTimeInstantiation.java
- [x] DuplicateTextInPreferencesXml.java
- [x] IssueRegistry.java
- [x] LintUtils.java

</details>

## nobnago
- [ ] ActivityTransitionAnimation.java
- [ ] Lookup.java
- [ ] MetaDB.java
- [ ] UIUtils.java
- [ ] Upgrade.java
- [ ] ViewAnimation.java

## Daniel Savard (many others)
- [ ] AnkiActivity.java

## AOSP/Google Apache - needs GPL
- [ ] AppCompatPreferenceActivity.java (timrae)
- [ ] CustomTabActivityHelper.java
- [ ] CustomTabsFallback.java
- [ ] CustomTabsHelper.java
- [ ] KeepAliveService.java
- [ ] SeekBarPreference.java (nonGoogle)

## John Dematis
- [ ] AudioClipField.java
- [ ] AudioRecordingField.java

## madhead (& others)
- [ ] BootService.java
- [ ] TimePreference.java

## Bruno Azevedo
- [ ] CardBrowserMySearchesDialog.java
- [ ] TagsDialog.java

## Valentin Rublack
- [ ] CardUtils.java

## timrae (Other licenses)
- [ ] ServiceConnection.java
- [ ] ServiceConnectionCallback.java

## Mike (Other licenses)
- [ ] ButtonItemAdapter.java

## Houssam Salem
- [ ] DeckRenameException.java
- [ ] ImportExportException.java
- [ ] MediaCheckDialog.java
- [ ] MediaSyncException.java
- [ ] NoteImporter.java
- [ ] TextImporter.java

## squeenix
- [ ] ModelBrowserContextMenu.java
- [ ] ModelEditorContextMenu.java

## Wojciech Śmigaj
- [ ] TtsParser.java

## Ritvij Kumar Sharma
- [ ] UserSubmittedException.java

## TESTS

Unlisted - grep code for the issue number

**Comment by TarekkMA** [comment]
Oops forgot to add to these files, here is where lint shines 😅  

should I add the copyright headers to my files?

**Comment by david-allison-1** [comment]
> Oops forgot to add to these files, here is where lint shines 😅
> 
> should I add the copyright headers to my files?

Definitely! Might as well cut the list before we figure out dealing with the harder aspects

**Comment by TarekkMA** [comment]
Should this lint also apply to tests? My `BundleUtilsTest` doesn't have a header but it's not in the list.

**Comment by david-allison-1** [comment]
> Should this lint also apply to tests? My `BundleUtilsTest` doesn't have a header but it's not in the list.

Yep - we're not tests currently

**Comment by mikehardy** [comment]
Where commit attribution is clear (which it is?) and the project is licensed uniformly (which it is?) it seems fairly cut and tried to this layman that we should be able to just put a copyright header in there with correct attribution personally.

**Comment by mikehardy** [comment]
Tim Rae #8682 
Mike Hardy #8683

**Comment by TarekkMA** [comment]
Should xml files have copyright headers as well  :thinking:

**Comment by mikehardy** [comment]
We have put some in where we took SVGs (which are XML) from other places. I think the answer is yes, XMLs are essentially code and should have copyright headers. But we're even farther from 100% on those unfortunately, so should probably be a second lint check, allowing us to at least see observable progress on the java side and taste some success earlier in the effort

**Comment by dae** [comment]
@mikehardy @david-allison still a problem?

**Comment by Rohitpatil990** [comment]
Hi! I’d like to work on this issue.

I plan to identify source files that are missing GPLv3-compatible copyright headers
and add the standard AnkiDroid header, following existing conventions.

I will avoid generated files and third-party code.

Please let me know if this approach is OK. Thanks!

**Comment by david-allison** [comment]
Probably best to leave this one to maintainers, we've got to discuss the legal side of this. 

It feels reasonable to assume that people submitted GPL compatible code under the `LICENSE`, but I want to be sure we have our ducks in a row before explicitly selecting a license on behalf of others. 

If it's not a reasonable legal action, we need a list of files which we'll want to reimplement in a compliant manner
